site/_posts/2021-05-11-manageiq-perf-results.md

```diff
@@ -0,0 +1,86 @@
+
+### Measurement configuration
+
+These measurements were taken on a vCenter dedicated to performance measurements with the following configuration:
+* Hypervisor:		VMware ESXi, 6.7.0, 15160138
+* Model:			IBM System X iDataPlex dx360 M4 Server -[7912AC1]-
+* Processor Type:		Intel(R) Xeon(R) CPU E5-2665 0 @ 2.40GHz
+* 14 ESX hosts: 		16 cores (32 logical processors), 256 GB Total Memory
+
+The podified measurements were taken on an OCP cluster with the following characteristics:
+* OCP 4.6.15
+* Kasparov branch
+* 3 OCP masters (8 core, 32 GB RAM)
+* 4 workers (8 core, 32 GB RAM)
+
+The appliance measurements were taken on a single VM running on the same vSphere cluster, with 16 cores, 64 GB RAM and a separate 200 GB hard disk.
+
+ManageIQ was configured with all server roles enabled, except Git Repositories Owner.  C & U Collection (metric collection) was enabled for all clusters and all datastores.
+
+For these measurements, up to 13 providers were added in a single, default zone:
+* 1 Amazon Web Service cloud provider
+* 1 Azure cloud provider
+* 1 VMware infrastructure provider
+* 10 additional VMware infrastructure providers simulated with the vCenter Server Simulator (VCSIM) tool, which is an unsupported tool available for free from VMware.  Each simulated vCenter runs on a separate VM.  
+
+### Steady state CPU utilization
+<img src="/assets/images/blog/PodSteadyStateCPU.png" width="250"></a>
+![](/assets/images/blog/PodSteadyStateCPU.png)
+
+The figure above shows the average process/pod CPU usage over a 2 hour period with the VMware provider running.  The appliance results (in blue) show the process-level CPU utilization based on data from the Linux ps command.   The podified results (in orange) show the pod-level CPU utilization from prometheus/grafana.  For most worker pods, CPU utilization in podified mode is consistently 8-9% higher than appliance mode.  In investigating this difference, the development team found that the extra overhead was caused by the liveness probe.  [Rewriting and optimizing the liveness check](https://github.com/ManageIQ/manageiq/pull/688) resulted in a 20x speedup.  The podified results with the optimized liveness check show up as gray bars in the figure.  The worker pods each show an 8-9% reduction in steady state CPU utilization.  This performance improvement are included in the Lasker branch. 
+
+The figure above also shows that the orchestrator CPU usage is lower in podified mode.  This is because the orchestrator pod doesn’t have to do the extra work of collecting worker CPU and memory usage, tracking worker heartbeats, and managing resource limits as it does in appliance mode.  For podified mode, this extra work is handled by Kubernetes.
+
+ <img src="/assets/images/blog/TotalSteadyStateCPU.png" width="250"></a>
+![](/assets/images/blog/TotalSteadyStateCPU.png)

```

Comment:
as mentioned above, we can probably drop this line if the `<img src` line works.